pocs/dome/abstract_serial_dome.py

```diff
@@ -0,0 +1,81 @@
+from pocs import dome
+from pocs.utils import error
+from pocs.utils import rs232
+
+
+class AbstractSerialDome(dome.AbstractDome):
+    """Abstract base class for controlling a dome via a serial connection.
+
+    Takes care of a single thing: configuring the connection to the device.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize an AbstractSerialDome.
+
+        Creates a serial connection to the port indicated in the config.
+        """
+        super().__init__(*args, **kwargs)
+
+        # Get config info, e.g. which port (e.g. /dev/ttyUSB123) should we use?
+        cfg = self._dome_config
+        self._port = cfg.get('port')
+        if not self._port:
+            msg = 'No port specified in the config for dome: {}'.format(cfg)
+            self.logger.error(msg)
+            raise error.DomeNotFound(msg=msg)
+
+        baudrate = int(cfg.get('baudrate', 9600))
+
+        # Setup our serial connection to the given port.
+        self.ser = None
+        try:
+            self.ser = rs232.SerialData(port=self._port, baudrate=baudrate)
+        except Exception as err:
+            raise error.DomeNotFound(err)
+
+    def __del__(self):
+        try:
+            if self.ser:
+                self.ser.disconnect()
+        except NameError:
+            pass
+
+    @property
+    def is_connected(self):
+        """True if connected to the hardware or driver."""
+        if self.ser:
+            return self.ser.is_connected
+        return False
+
+    def connect(self):
+        """ Connects to the device via the serial port (`self._port`)
```

Comment:
I still don't like advertising a private variable in the docstring. But I still kind of think the port shouldn't be private anyway.

Reply:
OK, will move it to public.

Reply:
Though I assume you agree that port shouldn't be changed, right?